crates/workers/src/lib.rs

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2024-2025 The Spice.ai OSS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#![allow(clippy::missing_errors_doc)]
+#![allow(clippy::missing_panics_doc)]
+#![allow(clippy::doc_markdown)]
+
+use async_trait::async_trait;
+use secrecy::SecretString;
+use serde_json::Value;
+use snafu::Snafu;
+use std::borrow::Cow;
+use std::collections::HashMap;
+use std::sync::Arc;
+
+pub mod router;
+pub use router::RouterModel;
+
+/// Error types for worker operations
+#[derive(Debug, Snafu)]
+pub enum Error {
+    #[snafu(display("Unsupported worker type: {worker_type}"))]
+    UnsupportedWorkerType { worker_type: String },
+
+    #[snafu(display("Invalid worker configuration: {message}"))]
+    InvalidWorkerConfig { message: String },
+}
+
+/// Result type for worker operations
+pub type Result<T, E = Error> = std::result::Result<T, E>;
+
+/// Workers that implement the [`SpiceWorker`] trait.
+#[async_trait]
+pub trait SpiceWorker: Sync + Send {
+    /// Get the source of the worker
+    fn from(&self) -> Cow<'_, str>;
+
+    /// Get the name of the worker
+    fn name(&self) -> Cow<'_, str>;
+
+    /// Get the role of the worker
+    fn role(&self) -> Cow<'_, str>;
+
+    /// Get the description of the worker, if any
+    fn description(&self) -> Option<Cow<'_, str>>;
+
+    /// Get the parameters of the worker
+    fn params(&self) -> &HashMap<String, Value>;
+
+    /// Load the worker's resources
+    async fn load(&self, _params: Arc<HashMap<String, SecretString>>) -> Result<()> {
+        Ok(())
+    }
+}
+
+/// Registry for managing workers
+pub struct WorkerRegistry {
```

Comment:
This doesn't appear to be used yet? Intentional?

Reply:
Correct, I removed it initially in #4893 but it must have been dragged in.